src/DiffSharp.Core/Model.fs

```diff
@@ -491,12 +491,21 @@ type ModelBase() =
 
 
 /// <summary>Represents a model, primarily a collection of named parameters and sub-models and a function governed by them.</summary>
-[<AbstractClass>]
-type Model<'In, 'Out>() =
+// [<AbstractClass>]
+type Model<'In, 'Out>(?f:'In->'Out, ?parameters: seq<Parameter>, ?buffers: seq<Parameter>, ?models: seq<obj>) =
```

Comment:
Should that `obj` be `ModelBase`?

Reply:
This design was to improve ease of use for `Model.addModel` etc. but I think it does make sense to get rid of this `obj` and use the correct type.

![image](https://user-images.githubusercontent.com/5365982/160290243-f8656587-b37f-49c7-8362-83c93165dd51.png)

Reply:
Fixed in a subsequent commit